src/main/java/org/kairosdb/core/http/rest/QueryAutocompleter.java

```diff
@@ -0,0 +1,49 @@
+package org.kairosdb.core.http.rest;
+
+import com.google.common.collect.SetMultimap;
+import org.kairosdb.core.datastore.QueryMetric;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class QueryAutocompleter {
+    private static final Logger logger = LoggerFactory.getLogger(QueryAutocompleter.class);
+
+    public void complete(QueryMetric query) {
+        completeMetricTag(query);
+    }
+
+    private void completeMetricTag(QueryMetric query) {
+        final SetMultimap<String, String> tags = query.getTags();
+        final Set<String> keys = tags.get("key");
+        final Set<String> metrics = new HashSet<>();
+        for (final String key : keys) {
+            try {
+                final String metric = extractMetricName(key);
+                if (isWildcard(metric)) {
```

Comment:
should usually not happen, but this might blow up with an NPE when `extractMetricName()`  returns `null`

Reply:
I can't imagine a situation when result of `extracMetricName` would be `null`. AFAIK `String::split` returns an array where every object is not null.

Reply:
'extracMetricName' returns null when inout parameter 'key' is empty or null

Reply:
lol, I'm dumb 🤦‍♂️thanks